openwebstart/src/main/java/com/openwebstart/proxy/WebStartProxySelector.java

```diff
@@ -55,7 +59,15 @@ private ProxyProvider createProvider(final DeploymentConfiguration config) {
 
     @Override
     public void connectFailed(final URI uri, final SocketAddress sa, final IOException ioe) {
-        DialogFactory.showErrorDialog(Translator.getInstance().translate("proxy.error.connectionFailed", uri), ioe);
-        JNLPRuntime.exit(-1);
+            assert proxyList != null && proxyList.size() != 0;
+
+            final String currentSocketAddress = sa != null ? sa.toString() : "DIRECT";
```

Comment:
Can you please rename `sa` to `proxyAddress` because it is the value of proxy.address() which is passed in the second argument

Reply:
ok